lib/charms/kratos/v0/kratos_registration_webhook.py

```diff
@@ -64,72 +59,49 @@ def deserialize_bool(v: str | bool) -> bool:
     return v
 
 
-def deserialize_model(v: Union[BaseModel, str], info: ValidationInfo) -> BaseModel:
-    if isinstance(v, BaseModel):
-        return v
-
-    return info.context["self"].model_fields[info.field_name].annotation(**from_json(v))
-
-
-def deserialize_optional_model(
-    v: Union[BaseModel, str], info: ValidationInfo
-) -> Optional[BaseModel]:
-    if v == "":
-        return None
-
-    if isinstance(v, BaseModel):
-        return v
-
-    t = info.context["self"].model_fields[info.field_name].annotation
-    for annotation in get_args(t):
-        if annotation is not type(None):
-            return annotation(**from_json(v))
-
-
 SerializableBool = Annotated[
     StrictBool,
     PlainSerializer(lambda v: str(v), return_type=str),
     BeforeValidator(deserialize_bool),
 ]
 
 
-SerializableModel = Annotated[
-    TypeVar("BaseModelType", bound=BaseModel),
-    PlainSerializer(lambda v: v.model_dump_json(), return_type=str),
-    BeforeValidator(deserialize_model),
-]
-
-
-OptionalSerializableModel = Annotated[
-    Optional[TypeVar("BaseModelType", bound=BaseModel)],
-    PlainSerializer(lambda v: v.model_dump_json() if v else "", return_type=str),
-    BeforeValidator(deserialize_optional_model),
-]
-
-
-class ResponseConfig(BaseModel):
-    ignore: SerializableBool
-    parse: SerializableBool
-
-
-class _AuthConfig(BaseModel):
-    name: str = "Authorization"
-    value: str
-    in_: str = Field(default="header", alias="in")
-
-
-class AuthConfig(BaseModel):
-    type: str = Field(default="api_key")
-    config: SerializableModel[_AuthConfig]
-
-
 class ProviderData(BaseModel):
     url: str
     body: str
     method: str
-    emit_analytics_event: SerializableBool
-    response: SerializableModel[ResponseConfig]
-    auth: OptionalSerializableModel[AuthConfig] = None
+    emit_analytics_event: SerializableBool = False
+    response_ignore: SerializableBool
+    response_parse: SerializableBool
+    auth_type: str = Field(default="api_key")
+    auth_config_name: Optional[str] = Field(default="Authorization")
+    auth_config_value: Optional[str] = Field(default=None, exclude=True)
+    auth_config_value_secret: Optional[str] = None
+    auth_config_in: Optional[str] = Field(default="header")
+
+    @cached_property
+    def auth_enabled(self) -> bool:
+        return all(
+            k
+            for k in [
+                self.auth_type,
+                self.auth_config_name,
+                self.auth_config_value_secret or self.auth_config_value,
+                self.auth_config_in,
+            ]
+        )
```

Comment:
suggestion: feel free to ignore, this should work the same without the list comprehension, even with the `or`, I tried this in a python3 out of curiosity
```suggestion
        return all(
            [
                self.auth_type,
                self.auth_config_name,
                self.auth_config_value_secret or self.auth_config_value,
                self.auth_config_in,
            ]
        )
```